client/my-sites/upgrades/cart/cart-body.jsx

```diff
@@ -1,50 +1,56 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var CartItems = require( './cart-items' ),
-	CartCoupon = require( './cart-coupon' ),
-	CartTotal = require( './cart-total' );
-
-var CartBody = React.createClass( {
-	propTypes: {
-		collapse: React.PropTypes.bool
-	},
-	getDefaultProps: function() {
-		return {
-			collapse: false,
-			showCoupon: false
-		};
-	},
-
-	render: function() {
-		if ( ! this.props.cart.hasLoadedFromServer ) {
-			return <div className="cart-body">{ this.translate( 'Loading…', { context: 'Upgrades: Loading cart' } ) }</div>;
+import CartItems from './cart-items';
+import CartCoupon from './cart-coupon';
+import CartTotal from './cart-total';
+
+class CartBody extends Component {
+	static propTypes = {
+		collapse: PropTypes.bool
+	};
+
+	static defaultProps = {
+		collapse: false,
+		showCoupon: false
+	};
+
+	optionalCoupon() {
+		if ( ! this.props.showCoupon ) {
+			return;
+		}
+
+		return <CartCoupon cart={ this.props.cart } />;
+	}
+
+	render() {
+		const { translate, cart } = this.props;
+
+		if ( ! cart.hasLoadedFromServer ) {
+			return <div className="cart-body">
+				{ translate( 'Loading…', { context: 'Upgrades: Loading cart' } ) }
```

Comment:
Yes.

All the comments are gathered together and translators see them all at once, so it's helpful for the translating itself, but also makes it really easy to recognize the rarer situations when it should have been a context.

So go nuts with comments, and be really conservative with contexts.